### PR TITLE
New implementation

### PR DESCRIPTION
**Pull request by extremeheat**
now a copy of minecraft-data but with some APIs to modify the data

### CODE REVIEW

**Comment by nickelpro** [comment]
I don't understand deleting registry.js

Maybe I'm misunderstanding what we're trying to do here. But to me, pris-reg should provide the _facilities_ to create registries. It should not actually be _creating_ those registries. Let mineflayer, bedrock-provider, and other consumers take that step, using the facilities provided here. This also solves this issue of having pris-reg be versioned, there's no reason for it to be versioned if its just a facility provider.

I'll PR with what I mean and we can compare notes

**Comment by rom1504** [comment]
I think the design makes sense

the problem I see with this specific implementation is that this will mutate the cached data loaded in minecraft data package hence not possible to have 2 bots in the same process with the same version but 2 servers providing different biomes

**Comment by extremeheat** [comment]
sure, I was thinking per rom's comment that he wanted an API directly like mc-data. So I couldn't think of a better way than just patching copy+patching ontop of mc-data exported data (though a ES6 Proxy could keep the old API around). But I agree for the namesake it does seem a bit odd.

As for mutating, it erases the data before writing new things so that shouldn't be an issue,
```js
    data.biomes = []
    data.biomesByName = {}
    data.biomesArray = []
```

**Comment by rom1504** [comment]
> Maybe I'm misunderstanding what we're trying to do here. But to me, pris-reg should provide the facilities to create registries. It should not actually be creating those registries. Let mineflayer, bedrock-provider, and other consumers take that step, using the facilities provided here. This also solves this issue of having pris-reg be versioned, there's no reason for it to be versioned if its just a facility provider.

I think prismarine-registry should be mostly drop-in to replace mcdata. Regarding where to parse data, curious to see your implementation

**Comment by rom1504** [comment]
> As for mutating, it erases the data before writing new things so that shouldn't be an issue,

ah right, perfect then

File: lib/pc/index.js [hunk 1]
```diff
@@ -1,6 +1,24 @@
-const Registry = require('../registry')
+const nbt = require('prismarine-nbt')
 
-module.exports = (version) => ({
-  biomes: new Registry(),
-  recipes: new Registry()
+module.exports = (data) => ({
+  loadDimensionCodec (codec) {
```
Comment (rom1504) [inline lib/pc/index.js:4]: can you document this in the api.md / readme.md

File: lib/loader.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+module.exports = (mcData) => {
+  const data = {
```
Comment (rom1504) [inline lib/loader.js]: maybe let's simply loop over the object ?

Comment (rom1504) [reply lib/loader.js]: yeah that would remove the need to duplicate the api, which then decrease the cost of the extra package

Comment (extremeheat) [reply lib/loader.js]: updated to use Object.assign on a new object

**Comment by extremeheat** [comment]
one thing to keep in mind here is that the data sent in the login dimension codec is not the same schema as the data in minecraft-data

**Comment by rom1504** [comment]
CI is failing

**Comment by extremeheat** [comment]
fixed

**Comment by rom1504** [comment]
let's go with that for now, see how things work

we can always migrate to another solution later on @nickelpro

**Pull request merged**